### PR TITLE
tests: run tiobe tics tool in a nightly workflow

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change introduces a new github actions workflow to run the analisys tool (Tiobe tics server)

The workflow has been tested in my forked project. The only step that is failing is the last one because the integration with tics server has to be done from snapd main project (not from the fork).

The data produced is going to be displayed in https://canonical.tiobe.com/tiobeweb/TICS/TqiDashboard.html#axes=Project(snapd),Sub()&metric=tqi

Next step is to integrate the analysis as part of the pr pipeline, including a quality gate.

### CODE REVIEW

File: .github/workflows/nightly.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  schedule:
+    - cron: '30 0 * * *'
+
+jobs:
+
+  tics:
+    runs-on: ubuntu-22.04
+    env:
+      GOPATH: ${{ github.workspace }}
+      # Set PATH to ignore the load of magic binaries from /usr/local/bin And
```
Comment (ernestl) [inline .github/workflows/nightly.yaml:14]: And -> and

File: .github/workflows/nightly.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        # is relevant for go's import system.
+        path: ./src/github.com/snapcore/snapd
+
+    - name: Download Debian dependencies
+      run: |
+          sudo apt clean
+          sudo apt update
+          sudo apt build-dep -y ${{ github.workspace }}/src/github.com/snapcore/snapd
+
+    # golang latest ensures things work on the edge
```
Comment (ernestl) [inline .github/workflows/nightly.yaml:43]: 1.18 is not the latest , maybe reword?

**Review by ernestl — APPROVED** [review]
Thanks, just 2 nitpicks

**Review by andrewphelpsj — APPROVED** [review]
Looks good, thank you!

File: .github/workflows/nightly.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply .github/workflows/nightly.yaml]: updated

File: .github/workflows/nightly.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply .github/workflows/nightly.yaml]: updated, thanks for the review
